### PR TITLE
Update CODEOWNERS with new Platform team

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Update CODEOWNERS with new Platform team. Mostly for core Go checks, invoke tasks, omnibus. Added a "catchall" rule for Platform for top-level files and other misc directories.

Also small update to ownership of Burrito and APM Agent teams.

### Motivation

New Platform team.

### Additional Notes

Some code ownership may be up for debate, happy to discuss if so.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -17,7 +19,7 @@
 /cmd/process-agent/                     @DataDog/burrito
 /cmd/system-probe/                      @DataDog/burrito
 
-/docs/                                  @DataDog/baklava @DataDog/agent-core
+/docs/                                  @DataDog/baklava @DataDog/agent-platform
```
Comment (albertvaka) [inline .github/CODEOWNERS]:
```suggestion
/docs/                                  @DataDog/baklava @DataDog/agent-platform
/docs/agent/                            @DataDog/baklava @DataDog/agent-core 
/docs/dogstatsd/                        @DataDog/baklava @DataDog/agent-core 
/docs/trace-agent/                      @DataDog/baklava @DataDog/apm-agent 
/docs/cluster-agent/                    @DataDog/baklava @DataDog/container-integrations
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -53,9 +61,14 @@
 /Dockerfiles/                           @DataDog/container-integrations
 /docs/trace-agent                       @DataDog/apm-agent
```
Comment (albertvaka) [inline .github/CODEOWNERS]:
```suggestion
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -9,6 +9,8 @@
 # Rules are matched bottom-to-top, so one team can own subdirectories
 # and another the rest of the directory.
 
+/                                       @DataDog/agent-platform
+
```
Comment (albertvaka) [inline .github/CODEOWNERS]:
```suggestion

/rtloader/                              @DataDog/agent-core
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -17,7 +19,7 @@
 /cmd/process-agent/                     @DataDog/burrito
 /cmd/system-probe/                      @DataDog/burrito
 
-/docs/                                  @DataDog/baklava @DataDog/agent-core
+/docs/                                  @DataDog/baklava @DataDog/agent-platform
```
Comment (KSerrania) [inline .github/CODEOWNERS]: I'd give agent-core ownership of the `/docs/dev/checks/` directory, as it contains rtloader-related docs. 
```suggestion
/docs/dev/checks/                       @DataDog/baklava @DataDog/agent-core

```

File: .github/CODEOWNERS (hunk 3 above)
Comment (olivielpeau) [reply .github/CODEOWNERS]: (good catch, I'll add this a bit below to preserve alphabetical ordering though)

Comment (olivielpeau) [reply .github/CODEOWNERS]: done

**Review by albertvaka — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**